airbyte-integrations/base-java/src/main/java/io/airbyte/integrations/javabase/IntegrationCliParser.java

```diff
@@ -0,0 +1,191 @@
+
+    switch (command) {
+      case SPEC -> {
+        // no args.
+      }
+      case CHECK, DISCOVER -> options.addOption(Option
+          .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+      case READ -> {
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_SCHEMA_KEY).desc(JavaBaseConstants.ARGS_SCHEMA_DESC).hasArg(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_STATE_KEY).desc(JavaBaseConstants.ARGS_PATH_DESC).hasArg(true).build());
+      }
+      case WRITE -> {
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_CONFIG_KEY).desc(JavaBaseConstants.ARGS_CONFIG_DESC).hasArg(true).required(true).build());
+        options.addOption(Option
+            .builder().longOpt(JavaBaseConstants.ARGS_SCHEMA_KEY).desc(JavaBaseConstants.ARGS_SCHEMA_DESC).hasArg(true).build());
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + command);
+    }
+
+    final CommandLine parsed = runParse(options, args, command);
+    Preconditions.checkNotNull(parsed);
+    final Map<String, String> argsMap = new HashMap<>();
+    for (final Option option : parsed.getOptions()) {
+      argsMap.put(option.getLongOpt(), option.getValue());
+    }
+
+    switch (command) {
+      case SPEC -> {
+        return IntegrationConfig.spec();
+      }
+      case CHECK -> {
+        return IntegrationConfig.check(argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY));
+      }
+      case DISCOVER -> {
+        return IntegrationConfig.discover(argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY));
+      }
+      case READ -> {
+        return IntegrationConfig.read(
+            argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_SCHEMA_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_STATE_KEY));
+      }
+      case WRITE -> {
+        return IntegrationConfig.write(
+            argsMap.get(JavaBaseConstants.ARGS_CONFIG_KEY),
+            argsMap.get(JavaBaseConstants.ARGS_SCHEMA_KEY));
+      }
+      default -> throw new IllegalStateException("Unexpected value: " + command);
+    }
+  }
+
+  private static CommandLine runParse(Options options, String[] args, Command command) {
+    final CommandLineParser parser = new DefaultParser();
+    final HelpFormatter helpFormatter = new HelpFormatter();
+
+    try {
+      return parser.parse(options, args);
+    } catch (ParseException e) {
+      LOGGER.error(e.toString());
+      helpFormatter.printHelp(command.toString().toLowerCase(), options);
+      return null;
```

Comment:
nit: slightly prefer to throw an exception or assert here as opposed to outside. Maybe `runParse` could just return the args map and all of the map creation can happen within the `try`?

Reply:
oh. that's a mistake left over from an old version. i'll throw here and not return null.